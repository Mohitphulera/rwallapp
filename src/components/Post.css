.card {
    margin-bottom: 1rem;
    margin-top: 1rem;
    opacity: 0.9;
}

.card-header {
    padding-left: 1.5rem;
    padding-top: 1rem;
    padding-bottom: 1rem;
}

.card-footer-item {
    justify-content: flex-start !important;
    padding: 1rem 1.5rem !important;
}