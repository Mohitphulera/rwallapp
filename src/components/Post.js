import React, { Component } from 'react'
import moment from 'moment'
import './Post.css'

export default class Post extends Component {
    render() {
        const {
            post: {
                author: {
                    username
                },
                message,
                timestamp
            }
        } = this.props
        return (
            <div className="card">
                <div className="card-header">
                    <div className="content">
                        <div className="media-content">
                            <p className="title is-5">{username}</p>
                        </div>
                    </div>
                </div>
                <div className="card-content">
                    <div className="content">
                        <div data-testid="message"className="is-size-5">{message}</div>
                    </div>
                </div>
                <div className="card-footer">
                    <div className="card-footer-item">
                        <time className="is-size-6">{moment(timestamp).utc().calendar()}</time>
                    </div>
                </div>
            </div>
        )
    }
}