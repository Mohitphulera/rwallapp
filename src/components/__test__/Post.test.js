import React from 'react'
import ReactDom from 'react-dom'
import Post from './../Post'
import {isTSAnyKeyword} from  '@babel/types'

import renderer from "react-test-renderer"

it("renders without crashing",()=> {
    const div = document.createElement("div");
    const post={author:{username:"mohit"},message:"hello i am there",timestamp:"2020-09-26T17:09:30.715855Z"}
    ReactDom.render(<Post post={post}/>,div)

})


it("matches snapshot & renders correctly",() => {
    const post={author:{username:"mohit"},message:"hello i am there",timestamp:"2020-09-26T17:09:30.715855Z"}
    const tree=renderer.create(<Post post={post}/>).toJSON();
    expect(tree).toMatchSnapshot();
})