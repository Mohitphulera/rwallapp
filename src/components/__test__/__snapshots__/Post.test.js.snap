// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot & renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <div
      className="content"
    >
      <div
        className="media-content"
      >
        <p
          className="title is-5"
        >
          mohit
        </p>
      </div>
    </div>
  </div>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <div
        className="is-size-5"
        data-testid="message"
      >
        hello i am there
      </div>
    </div>
  </div>
  <div
    className="card-footer"
  >
    <div
      className="card-footer-item"
    >
      <time
        className="is-size-6"
      >
        Last Saturday at 10:39 PM
      </time>
    </div>
  </div>
</div>
`;
